collated/functional/jingyinno.md

```diff
@@ -0,0 +1,1013 @@
+    private static final String VACANT_STYLE_CLASS = "vacant";
+    private static final String FXML = "VenueTable.fxml";
+
+    private ArrayList<TableColumn<ArrayList<String>, String>> columns;
+    @FXML
+    private TableView venueTable;
+    @FXML
+    private TableColumn<ArrayList<String>, String> roomId;
+    @FXML
+    private TableColumn<ArrayList<String>, String> eightAm;
+    @FXML
+    private TableColumn<ArrayList<String>, String> nineAm;
+    @FXML
+    private TableColumn<ArrayList<String>, String> tenAm;
+    @FXML
+    private TableColumn<ArrayList<String>, String> elevenAm;
+    @FXML
+    private TableColumn<ArrayList<String>, String> twelvePm;
+    @FXML
+    private TableColumn<ArrayList<String>, String> onePm;
+    @FXML
+    private TableColumn<ArrayList<String>, String> twoPm;
+    @FXML
+    private TableColumn<ArrayList<String>, String> threePm;
+    @FXML
+    private TableColumn<ArrayList<String>, String> fourPm;
+    @FXML
+    private TableColumn<ArrayList<String>, String> fivePm;
+    @FXML
+    private TableColumn<ArrayList<String>, String> sixPm;
+    @FXML
+    private TableColumn<ArrayList<String>, String> sevenPm;
+    @FXML
+    private TableColumn<ArrayList<String>, String> eightPm;
+
+    public VenueTable(ObservableList<ArrayList<String>> schedules) {
+        super(FXML);
+        venueTable.setItems(schedules);
+        initializeColumns();
+        initializeTableColumns();
+        roomId.setMinWidth(100);
+        roomId.setMaxWidth(100);
+        venueTable.setColumnResizePolicy(TableView.CONSTRAINED_RESIZE_POLICY);
+    }
+
+    /**
+     * Initializes columns
+     */
+    private void initializeColumns() {
+        columns = new ArrayList<>();
+        columns.add(roomId);
+        columns.add(eightAm);
+        columns.add(nineAm);
+        columns.add(tenAm);
+        columns.add(elevenAm);
+        columns.add(twelvePm);
+        columns.add(onePm);
+        columns.add(twoPm);
+        columns.add(threePm);
+        columns.add(fourPm);
+        columns.add(fivePm);
+        columns.add(sixPm);
+        columns.add(sevenPm);
+        columns.add(eightPm);
+    }
+
+    /**
+     * Initializes table columns
+     */
+    private void initializeTableColumns() {
+        for (int i = 0; i < columns.size(); i++) {
+            final int j = i;
+            columns.get(i).setCellValueFactory(cellData -> new SimpleStringProperty(cellData.getValue().get(j)));
+            columns.get(i).impl_setReorderable(false);
+            if (j != 0) {
+                columns.get(i).setMinWidth(75);
+                columns.get(i).setMaxWidth(200);
+            }
+
+        }
+    }
+    /**
+     * Sets the command box style to indicate a vacant or occupied room.
+     */
+    public void setStyle() {
+        for (int i = 0; i < columns.size(); i++) {
+            columns.get(i).setCellFactory(column -> {
+                return new TableCell<ArrayList<String>, String>() {
```

Comment:
Usually it's okay to use functions like this, but this is CS2103 and we want to avoid things being nested like this. So here I would recommend you to extract the `TableCell` constructor out if possible.